### PR TITLE
[VOTE before merging] Update the givernance

### PR DESCRIPTION
**Pull request by alabulei1**
1. Re-org the GOVERNANCE.md
2. Add docs/CONTRIBUTOR_LADDER.md
3. Add docs/CONTRIBUTING.md

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/). Here are my reviews of changed source code files in this PR.

------

File: docs/CONTRIBUTOR_LADDER.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+Committers are responsible for a "specific area." This can be a specific code directory, driver, chapter of the docs, test job, event, or other clearly-defined project component that is smaller than an entire repository or subproject. Most often it is one or a set of directories in one or more Git repositories. The "specific area" below refers to this area of responsibility.
+
+Committers have all the rights and responsibilities of a reviewer, plus:
+
+* Responsibilities include:
+    * Reviewing most Pull Requests against their specific areas of responsibility
+    * Reviewing at least 15 PRs per year
+    * Helping other contributors become Committers
+* Requirements:
+    * Experience as a Reviewer for at least 6 months
```
Comment (juntao) [inline docs/CONTRIBUTOR_LADDER.md:102]: Let's change that to 3 months?

File: docs/CONTRIBUTOR_LADDER.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+Description: A Reviewer is an established contributor who regularly participates in the project. Reviewers have privileges in both project repositories and elections, and as such are expected to act in the interests of the whole project.
+
+A Reviewer must meet the responsibilities and has the requirements of a Contributor, plus:
+
+* Responsibilities include:
+    * Continues to contribute regularly, as demonstrated by having at least 30 GitHub contributions per year, as shown by Devstats. 
+* Requirements:
+    * Must have successful contributions to the project, including at least one of the following:
+        * 5 accepted PRs,
```
Comment (juntao) [inline docs/CONTRIBUTOR_LADDER.md:69]: I think 2 would be sufficient. Raising an PR and getting accepted is a high bar.

File: docs/CONTRIBUTOR_LADDER.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+Committers are responsible for a "specific area." This can be a specific code directory, driver, chapter of the docs, test job, event, or other clearly-defined project component that is smaller than an entire repository or subproject. Most often it is one or a set of directories in one or more Git repositories. The "specific area" below refers to this area of responsibility.
+
+Committers have all the rights and responsibilities of a reviewer, plus:
+
+* Responsibilities include:
+    * Reviewing most Pull Requests against their specific areas of responsibility
+    * Reviewing at least 15 PRs per year
+    * Helping other contributors become Committers
+* Requirements:
```
Comment (juntao) [inline docs/CONTRIBUTOR_LADDER.md:101]: Committer needs to contribute code. I think we can make it 5 PRs with code changes (not just docs or typo fixes).

File: docs/CONTRIBUTOR_LADDER.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+1. Any current Maintainer may nominate a current Reviewer to become a new Maintainer, by opening a PR against the root of the WasmEdge/WasmEdge repo adding the nominee as an Approver in the OWNERS file.
+2. The nominee will add a comment to the PR testifying that they agree to all requirements of becoming a Maintainer.
+3. A 2/3 of the current Maintainers must then approve the PR.
+
+
+## Inactivity
+
+It is important for contributors to be and stay active to set an example and show commitment to the project. Inactivity is harmful to the project as it may lead to unexpected delays, contributor attrition, and a lost of trust in the project.
+
+* Inactivity is measured by:
```
Comment (juntao) [inline docs/CONTRIBUTOR_LADDER.md:159]: I think 12 months would be better.

File: docs/CONTRIBUTOR_LADDER.md (hunk 3 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:101]: Agreed

File: docs/CONTRIBUTOR_LADDER.md (hunk 4 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:159]: Agreed

File: docs/CONTRIBUTOR_LADDER.md (hunk 1 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:102]: Sure

**Comment by alabulei1** [comment]
Updated. @hydai @dm4 @q82419  @ibmibmibm Please take a look. Thanks.

File: docs/CONTRIBUTOR_LADDER.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    * [Maintainer](#maintainer)
+* [Inactivity](#inactivity)
+* [Involuntary Removal](#involuntary-removal-or-demotion)
+* [Stepping Down/Emeritus Process](#stepping-downemeritus-process)
+* [Contact](#contact)
+
+
+## Contributor Ladder
+
+Hello! We are excited that you want to learn more about our project contributor ladder! This contributor ladder outlines the different contributor roles within the project, along with the responsibilities and privileges that come with them. Community members generally start at the first levels of the "ladder" and advance up it as their involvement in the project grows.  Our project members are happy to help you advance along the contributor ladder.
```
Comment (hydai) [inline docs/CONTRIBUTOR_LADDER.md:17]: There are two spaces between `...the project grows.` and `Our project members...`

File: docs/CONTRIBUTOR_LADDER.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    * Trying out new releases
+    * Attending community events, like community meetings
+  
+
+### Contributor
+
+Description: A Contributor contributes directly to the project and adds value to it. Contributions need not be code. People at the Contributor level may be new contributors, or they may only contribute occasionally.
+
+* Responsibilities include:
+    * Follow the CNCF CoC
```
Comment (hydai) [inline docs/CONTRIBUTOR_LADDER.md:42]: Should we also add a link to the `CNCF CoC` here?

File: docs/CONTRIBUTOR_LADDER.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    * Follow the project contributing guide
```
Comment (hydai) [inline docs/CONTRIBUTOR_LADDER.md:43]: What's the link to `the project contributing guide`? Is it this one?

File: docs/CONTRIBUTOR_LADDER.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+The process of becoming a Committer is:
+
+1. The contributor is nominated by opening a PR against the appropriate repository by the existing reviewers, committers and maintainers, which adds their GitHub username to the OWNERS file for one or more directories.
+2. At least 1/2 members of the reviewers, committers, and maintainers that owns that repository or main directory, who are already Approvers, approve the PR.
+
+
+### Maintainer
+<!-- In the simplest and most common project structures, projects have a single pool of "maintainers" who are collectively responsible for the entire project.  This example defines that role for your project.  If your project has a more complex structure, see the list of specific maintainer roles you might want to define, below. -->
+<!--TODO: project leads to fill in exact details of this role for your project-->
```
Comment (hydai) [inline docs/CONTRIBUTOR_LADDER.md:122]: What's the intention behind keeping these comments here?

File: docs/CONTRIBUTOR_LADDER.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+
+* Responsibilities include:
+    * Reviewing at least 30 PRs per year, especially PRs that involve multiple parts of the project
+    * Mentoring new Reviewers
+    * Writing refactoring PRs
+    * Participating in CNCF maintainer activities
+    * Determining strategy and policy for the project
+    * Participating in, and leading, community meetings
+* Requirements
+    * Experience as a Reviewer for at least 3 months
```
Comment (hydai) [inline docs/CONTRIBUTOR_LADDER.md:136]: Should it be `Committer`?

File: docs/CONTRIBUTOR_LADDER.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+    * Demonstrates a broad knowledge of the project across multiple areas
+    * Is able to exercise judgment for the good of the project, independent of their employer, friends, or team
+    * Mentors other contributors
```
Comment (hydai) [inline docs/CONTRIBUTOR_LADDER.md:139]: Is this a requirement or a responsibility?

File: docs/CONTRIBUTOR_LADDER.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+* Additional privileges:
+    * Approve PRs to any area of the project
+    * Represent the project in public as a Maintainer
+    * Communicate with the CNCF on behalf of the project
+    * Have a vote in Maintainer decision-making meetings
+    
+
+Process of becoming a maintainer:
+
+1. Any current Maintainer may nominate a current Reviewer to become a new Maintainer, by opening a PR against the root of the WasmEdge/WasmEdge repo adding the nominee as an Approver in the OWNERS file.
```
Comment (hydai) [inline docs/CONTRIBUTOR_LADDER.md:150]: I would like to confirm what's the process of nominating a new maintainer.
Would it be:
1. Contributor -> Reviewer -> Committer -> Maintainer
or
2. Contributor -> Reviewer -> Maintainer

File: docs/GOVERNANCE.md [hunk 12]
```diff
@@ -2,31 +2,74 @@
...
+There are three levels of maintainers for WasmEdge. The WasmEdge maintainers oversee the overall
+project and its health. Committers focus on a single codebase, a group of related
+codebases, a service (e.g., a website), or project to support the other projects (e.g., marketing or
+community management). Reviewers help review the GitHub issues and PRs. See the [Contributor Ladder](./CONTRIBUTION_LADDER.md) for more detailed information on responsibilities.
+
+### Maintainers
+
+WasmEdge Maintainers have write access to the [WasmEdge](https://github.com/WasmEdge/WasmEdge) GitHub repo.
+They can merge their own patches or patches from others. The current maintainers
+can be found in [MAINTAINERS.md](./MAINTAINERS.md).  Maintainers collectively manage the project's
```
Comment (hydai) [inline docs/GOVERNANCE.md:52]: We don't use `MAINTAINERS.md`. All of the information is inside `OWNERS.md`. Do you want to create a new `MAINTAINERS.md` or use the existing one?

**Review by hydai — CHANGES_REQUESTED** [review]

File: docs/CONTRIBUTOR_LADDER.md (hunk 7 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:43]: We have one. I will add the link.

File: docs/CONTRIBUTOR_LADDER.md (hunk 5 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:17]: OK

File: docs/CONTRIBUTOR_LADDER.md (hunk 8 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:122]: I can delete it.

File: docs/CONTRIBUTOR_LADDER.md (hunk 9 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:136]: Let me check. Thanks.

File: docs/CONTRIBUTOR_LADDER.md (hunk 10 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:139]: Let me check

File: docs/GOVERNANCE.md (hunk 12 above)
Comment (alabulei1) [reply docs/GOVERNANCE.md:52]: We will use OWNER.md. I will change it.

File: docs/CONTRIBUTOR_LADDER.md (hunk 10 above)
Comment (alabulei1) [reply docs/CONTRIBUTOR_LADDER.md:139]: It's a requirement.

**Comment by alabulei1** [comment]
@hydai @dm4 @q82419 @ibmibmibm Please take a look and vote. Thanks.

**Review by hydai — APPROVED** [review]

**Review by q82419 — APPROVED** [review]

**Comment by hydai** [comment]
Vote passed. Merging it.

**Pull request merged**